From d72d5e8aeb7df99c55bdc438fb71fdeffd2bd1e5 Mon Sep 17 00:00:00 2001
From: Takashi Yano <takashi.yano@nifty.ne.jp>
Date: Mon, 9 May 2022 12:23:57 +0900
Subject: [PATCH 133/N] Cygwin: pty: Not to change code page of parent
 console.

- The recent commit "Cygwin: pty: Fix timing of creating invisible
  console." breaks the feature added by commit 72770148, which
  prevents pty from changing code page of parent console. This patch
  restores that.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/fhandler_console.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/fhandler_console.cc b/winsup/cygwin/fhandler_console.cc
index d2ed15a..e848236 100644
--- a/winsup/cygwin/fhandler_console.cc
+++ b/winsup/cygwin/fhandler_console.cc
@@ -1608,7 +1608,7 @@ fhandler_console::close ()
   if (con_ra.rabuf)
     free (con_ra.rabuf);
 
-  if (!have_execed)
+  if (!have_execed && !invisible_console)
     free_console ();
   return 0;
 }
-- 
2.9.0

